server/src/test/java/org/elasticsearch/action/admin/ReloadSecureSettingsIT.java

```diff
@@ -205,7 +206,10 @@ public void onResponse(NodesReloadSecureSettingsResponse nodesReloadResponse) {
                             assertThat(nodesMap.size(), equalTo(cluster().size()));
                             for (final NodesReloadSecureSettingsResponse.NodeResponse nodeResponse : nodesReloadResponse.getNodes()) {
                                 assertThat(nodeResponse.reloadException(), notNullValue());
-                                assertThat(nodeResponse.reloadException(), instanceOf(IOException.class));
+                                // Running in a JVM with a BouncyCastle FIPS Security Provider, decrypting the Keystore with the wrong
+                                // password can return a SecurityException if the DataInputStream can't be fully consumed
+                                assertThat(nodeResponse.reloadException(),
```

Comment:
I think we should set an additional system property when running in a fips jvm so tests can conditionalize checks like this.

Reply:
Then we should add whatever extra properties are necessary for the test to distinguish the two cases. What would happen if one jvm started throwing the other exception? The test would be out of date but we would have no idea there was a behavior change.